services/network/HttpClientImpl.cpp

```diff
@@ -222,9 +222,11 @@ namespace services
             else
             {
                 bodyReader.Emplace(ConnectionObserver::Subject().ReceiveStream(), *contentLength);
+                auto reader = infra::MakeContainedSharedObject(bodyReader->countingReader, bodyReaderAccess.MakeShared(bodyReader));
+                BodyReaderAvailable(reader);
 
                 if (HttpClient::IsAttached())
-                    Observer().BodyAvailable(infra::MakeContainedSharedObject(bodyReader->countingReader, bodyReaderAccess.MakeShared(bodyReader)));
+                    Observer().BodyAvailable(std::move(reader));
```

Comment:
Please move this into BodyReaderAvailable, and have descendant implementations call their inherited BodyReaderAvailable

Reply:
Good suggestion! done!